### PR TITLE
HW/CPU: Refactor to class, move to System.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**

### CODE REVIEW

File: Source/Core/Core/HW/CPU.cpp [hunk 1]
```diff
@@ -190,57 +166,57 @@ void Run()
...
-    s_state_cpu_idle_cvar.wait(state_lock);
+    m_state_cpu_idle_cvar.wait(state_lock);
   }
 
   RunAdjacentSystems(false);
   FlushStepSyncEventLocked();
 }
 
-bool IsStepping()
+bool CPUManager::IsStepping()
```
Comment (lioncash) [inline Source/Core/Core/HW/CPU.cpp:198]:
```suggestion
bool CPUManager::IsStepping() const
```

File: Source/Core/Core/HW/CPU.cpp [hunk 2]
```diff
@@ -190,57 +166,57 @@ void Run()
...
 {
-  return s_state == State::Stepping;
+  return m_state == State::Stepping;
 }
 
-State GetState()
+State CPUManager::GetState()
```
Comment (lioncash) [inline Source/Core/Core/HW/CPU.cpp:203]:
```suggestion
State CPUManager::GetState() const
```

File: Source/Core/Core/HW/CPU.cpp [hunk 3]
```diff
@@ -190,57 +166,57 @@ void Run()
...
 {
-  return s_state;
+  return m_state;
 }
 
-const State* GetStatePtr()
+const State* CPUManager::GetStatePtr()
```
Comment (lioncash) [inline Source/Core/Core/HW/CPU.cpp:208]:
```suggestion
const State* CPUManager::GetStatePtr() const
```

**Comment by AdmiralCurtiss** [comment]
yup, fixed

**Review by lioncash — APPROVED** [review]

**Pull request merged**